web_programming/web_scrapping.py

```diff
@@ -0,0 +1,144 @@
+    except KeyError:
+        try:
+            title = soup.find("meta", attrs={"name": "title"})["content"]
+            if title:
+                meta_info["title"] = title
+        except KeyError:
+            meta_info["title"] = "Not mentioned"
+
+    try:
+        type_ = soup.find("meta", property="og:type")["content"]
+        if type_:
+            meta_info["type"] = type_
+    except KeyError:
+        try:
+            type_ = soup.find("meta", attrs={"name": "type"})["content"]
+            if type_:
+                meta_info["type"] = type_
+        except KeyError:
+            meta_info["type"] = "Not mentioned"
+
+    try:
+        description = soup.find("meta", property="og:description")["content"]
+        if description:
+            meta_info["description"] = description
+    except KeyError:
+        try:
+            description = soup.find("meta", attrs={"name": "description"})["content"]
+            if description:
+                meta_info["description"] = description
+        except KeyError:
+            meta_info["description"] = "Not mentioned"
+
+    try:
+        site_name = soup.find("meta", property="og:site_name")["content"]
+        if site_name:
+            meta_info["site_name"] = site_name
+    except KeyError:
+        try:
+            site_name = soup.find("meta", attrs={"name": "site_name"})["content"]
+            if site_name:
+                meta_info["site_name"] = site_name
+        except KeyError:
+            meta_info["site_name"] = "Not mentioned"
+
+    try:
+        image = soup.find("meta", property="og:image")["content"]
+        if image:
+            meta_info["image"] = image
+    except KeyError:
+        try:
+            image = soup.find("meta", attrs={"name": "image"})["content"]
+            if image:
+                meta_info["image"] = image
+        except KeyError:
+            meta_info["image"] = "Not mentioned"
+
+    try:
+        keywords = soup.find("meta", property="og:keywords")["content"]
+        if keywords:
+            meta_info["keywords"] = keywords
+    except KeyError:
+        try:
+            keywords = soup.find("meta", attrs={"name": "keywords"})["content"]
+            if keywords:
+                meta_info["keywords"] = keywords
+        except KeyError:
+            meta_info["keywords"] = "Not mentioned"
+
+    if meta_info["title"] == "Not mentioned":
+        title = soup.find("title").text
+        meta_info["title"] = title
+    print(meta_info)
+    meta_list = []
+    for key, value in meta_info.items():
+        l = []
+        l.insert(0, key)
+        l.append(value)
+        meta_list.append(l)
+
+    all_tags = []
+    allhrefs = set()
+    for a in soup.find_all("a", href=True):
+        if a["href"].find("https://") == -1:
+            allhrefs.add("https://machinelearningmastery.com/blog/" + a["href"])
+        else:
+            allhrefs.add(a["href"])
+    print(allhrefs)
+    hrefs = list(allhrefs)
```

Comment:
OPTIONAL: Is it interesting to sort the content that `set()` will leave in some random order?
```suggestion
    hrefs = sorted(allhrefs)
```

Reply:
There is no advantages of a sorted href list